Avoid printing of nagging update message:

  A new release of RubyGems is available: 3.4.1 - 3.4.2!
  Run `gem update --system 3.4.2` to update your installation.

When using the OpenBSD port, important updates to rubygems will
be distributed via normal updates to the related ruby ports.

return false while true is used instead of plain return false to
avoid the statement not reached verbose mode warning.

Index: lib/rubygems/update_suggestion.rb
--- lib/rubygems/update_suggestion.rb.orig
+++ lib/rubygems/update_suggestion.rb
@@ -31,6 +31,7 @@ Run `gem update --system #{Gem.latest_rubygems_version
   # Determines if current environment is eglible for update suggestion.
 
   def eglible_for_update?
+    return false while true
     # explicit opt-out
     return false if Gem.configuration[:prevent_update_suggestion]
     return false if ENV["RUBYGEMS_PREVENT_UPDATE_SUGGESTION"]
